lib/context.js

```diff
@@ -52,42 +60,65 @@ module.exports = async function getContext() {
-          },
-          commits: [
-            {
-              message: execSync('git log --format=%B -n 1 $BITBUCKET_COMMIT', {
-                env: { BITBUCKET_COMMIT: process.env.BITBUCKET_COMMIT },
-                encoding: 'utf-8',
-              }).trim(),
-              // Get the unix timestamp of the commit then convert it to
-              // an ISO8601 UTC string
-              //
-              // https://git-scm.com/docs/pretty-formats#Documentation/pretty-formats.txt-emctem
-              // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString
-              timestamp: new Date(
-                execSync('git show -s --format=%ct $BITBUCKET_COMMIT', {
-                  env: { BITBUCKET_COMMIT: process.env.BITBUCKET_COMMIT },
-                  encoding: 'utf-8',
-                }).trim() * 1000
-              ).toISOString(),
-              author: {
-                name: actor,
+      // BITBUCKET_PR_ID only set on pull request triggered builds
+      // this is how we determine if it's a `push` or `pull_request`
+      const eventName = process.env.BITBUCKET_PR_ID ? 'pull_request' : 'push';
+
+      switch (eventName) {
+        case 'push':
+          return {
+            ...baseContext,
+            eventName,
+            payload: {
+              repository: {
+                name: process.env.BITBUCKET_REPO_SLUG,
+                // TODO we should remove this in a future release
+                organization: process.env.BITBUCKET_WORKSPACE,
+              },
+              organization: {
+                login: process.env.BITBUCKET_WORKSPACE,
               },
+              commits: [
+                {
+                  message: execSync('git log --format=%B -n 1 $BITBUCKET_COMMIT', {
+                    env: { BITBUCKET_COMMIT: process.env.BITBUCKET_COMMIT },
+                    encoding: 'utf-8',
+                  }).trim(),
+                  // Get the unix timestamp of the commit then convert it to
+                  // an ISO8601 UTC string
+                  //
+                  // https://git-scm.com/docs/pretty-formats#Documentation/pretty-formats.txt-emctem
+                  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString
+                  timestamp: new Date(
+                    execSync('git show -s --format=%ct $BITBUCKET_COMMIT', {
+                      env: { BITBUCKET_COMMIT: process.env.BITBUCKET_COMMIT },
+                      encoding: 'utf-8',
+                    }).trim() * 1000
+                  ).toISOString(),
+                  author: {
+                    name: actor,
+                  },
+                },
+              ],
             },
-          ],
-        },
-      };
+          };
+        case 'pull_request':
+          return {
+            ...baseContext,
+            eventName,
+            payload: {
+              number: process.env.BITBUCKET_PR_ID,
```

Comment:
how about `parseInt`'ing this - to keep the payload closer resembling to GH's?

the GH fixture `pull_request.opened.json` shows `number: 2`. it's a [number in GH payloads](https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request), and a number in the Micro DB schema.

I wasn't entirely clear from PR #373 of how it's getting coerced, since I didn't see any changes to the [`pr` object](https://github.com/readmeio/micro/blob/47a9700b0/lib/db.js#L759) where `number` is assigned to `data.payload.number`

Reply:
Yeah that's not a bad idea! I actually modified the json schema stuff in the backend to coerce types now, so it matters less. But probably a good idea to be as consistent as possible.